libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/feedback/ui/CarGridFeedbackScreen.kt

```diff
@@ -65,16 +75,29 @@ class CarGridFeedbackScreen constructor(
     private fun buildItemList(options: List<CarFeedbackOption>): ItemList {
         val itemListBuilder = ItemList.Builder()
         for (option in options) {
-            val gridItem = GridItem.Builder()
+            val itemBuilder = GridItem.Builder()
                 .setTitle(option.title)
-                .setImage(option.icon, GridItem.IMAGE_TYPE_ICON)
                 .setOnClickListener { selectOption(option) }
-                .build()
-            itemListBuilder.addItem(gridItem)
+            val image = when (option.icon) {
+                is CarFeedbackIcon.Local -> option.icon.icon
+                is CarFeedbackIcon.Remote -> downloadedImages.getOrPut(option.icon.uri) {
+                    lifecycleScope.launch { downloadImage(option.icon.uri) }
+                    CarIcon.ERROR
```

Comment:
Why error? Do you just trigger the request here and meanwhile put the error?

Reply:
I decided to use error icon as default, because all of the feedback categories represent some kind of an error. For example an error on the map or an error during navigation. Also Android Auto will crash if you don't set an icon, so you have to set something while the image is being loaded.

Reply:
Agree that it shouldn't be an error unless it's _actionable_, how about finding a successful place holder?

Most icon download placeholders are "shimmers" these days.

![Screen Shot 2022-08-25 at 3 39 30 PM](https://user-images.githubusercontent.com/3021882/186781029-103e94e2-b3b7-44ea-b136-9d53323cfb66.png)

Reply:
Alright, I found out that each item can be put into a loading state, in this case it will show a loading indicator. There is one downside though, in this state it can't be made clickable. Thus I set a timeout of 3 seconds for downloading the icon. If download files, then we show the error icon. I added a video to the PR description.